src/iuse_actor.cpp

```diff
@@ -1328,11 +1328,9 @@ int salvage_actor::cut_up( player &p, item &it, item &cut ) const
     // Clean up before removing the item.
     remove_ammo( cut, p );
     // Original item has been consumed.
-    if( pos != INT_MIN ) {
-        p.i_rem( pos );
-    } else {
-        g->m.i_rem( p.posx(), p.posy(), &cut );
-    }
+    std::vector<item_comp> comps;
+    comps.push_back( item_comp( cut.typeId(), 1 ) );
```

Comment:
Wouldn't that remove *any* random item of that type within crafting distance, instead of the item that is *actually* cut?

Reply:
Yes, you are right.